src/jdk.jartool/share/classes/module-info.java

```diff
@@ -39,6 +39,11 @@
  * @toolGuide jar
  * @toolGuide jarsigner
  *
+ * @provides java.util.spi.ToolProvider
+ *         Pass {@code "jar"} as the name to
+ *         {@link java.util.spi.ToolProvider#findFirst ToolProvider.findFirst}
+ *         in order to obtain an instance of the tool.
```

Comment:
I'm not sure about the wording. It might be better to say that it provides a tool named "jar". Invoke findFirst("jar") to create an instance of this tool.

Reply:
What is "it" in "it provides..." ?

Reply:
Perhaps like this?

```java
/**
 * ...
 * @provides java.util.spi.ToolProvider
 *         Module {@code jdk.jartool} provides a tool named {@code "jar"}.
 *         Invoke {@link java.util.spi.ToolProvider#findFirst ToolProvider.findFirst("jar")}
 *         to create an instance of this tool.
 * ...
 */
```

Reply:
> Perhaps like this?
> 
> ```java
> /**
>  * ...
>  * @provides java.util.spi.ToolProvider
>  *         Module {@code jdk.jartool} provides a tool named {@code "jar"}.
>  *         Invoke {@link java.util.spi.ToolProvider#findFirst ToolProvider.findFirst("jar")}
>  *         to create an instance of this tool.
>  * ...
>  */
> ```

What about

`Module {@code jdk.jartool) provides the equivalent of command-line access to the {@code "jar"} tool`

Reply:
The focus should be to document the service specified in the `@provides` directive, and how to access to access an instance of the service.

Reply:
I think Jon's latest proposal combines all requirements with using better wording than my initial text.

Do you agree, @AlanBateman and @LanceAndersen?

Reply:
> I think Jon's latest proposal combines all requirements with using better wording than my initial text.
> 
> Do you agree, @AlanBateman and @LanceAndersen?

Yes, I think that sounds good